libs/crosstarget-utils/src/wasm/spawn.rs

```diff
@@ -1,10 +1,23 @@
 use std::future::Future;
 
+use futures::FutureExt;
+use tokio::sync::oneshot;
+use wasm_bindgen_futures::spawn_local;
+
 use crate::common::SpawnError;
 
-pub async fn spawn_if_possible<F>(future: F) -> Result<F::Output, SpawnError>
+pub fn spawn_if_possible<F>(future: F) -> impl Future<Output = Result<F::Output, SpawnError>>
 where
     F: Future + 'static,
 {
-    Ok(future.await)
+    let (sx, rx) = oneshot::channel::<F::Output>();
+    spawn_local(async move {
+        let result = future.await;
+        let _ = sx.send(result);
```

Comment:
fwiw you can also write `_ = sx.send(result)` here without `let` if you'd prefer